WeakAuras/GenericTrigger.lua

```diff
@@ -1582,6 +1582,38 @@ do
       mainSpeed, offSpeed = mainSpeedNew, offSpeedNew
     elseif casting and (event == "UNIT_SPELLCAST_INTERRUPTED" or event == "UNIT_SPELLCAST_FAILED") then
       casting = false
+    elseif event == "PLAYER_EQUIPMENT_CHANGED" and isAttacking then
+      local currentTime = GetTime()
+      mainSpeed, offSpeed = UnitAttackSpeed("player")
+      offSpeed = offSpeed or 0
+      local startEvent = false
+      if lastSwingMain then
+        lastSwingMain = currentTime
+        swingDurationMain = mainSpeed
+        mainSwingOffset = 0
+        timer:CancelTimer(mainTimer)
+        mainTimer = timer:ScheduleTimerFixed(swingEnd, mainSpeed, "main")
+        startEvent = true
+      end
+      if lastSwingOff then
```

Comment:
I'm not sure whether you consider that a weapon swap could be from e.g. a two-hander to two weapons. Or from a normal weapon to a ranged weapon.

Don't you need to check for what kind of weapons are equipped?

Reply:
From my tests switching from any weapon to another reset swing timer, e.g. having no offhand and equipping a shield reset main hand

Reply:
Yeah, but e.g. if you switch from a 2 weapons to one weapons, you still seem to schedule a timer for the offhand?

Reply:
if you had a 2H weapon lastSwingOff should be nil

Reply:
previously 2 weapons,
now one weapon

Reply:
What if I disconnect in a fight and log back in, will the swing timer still be accurate for the next swing or only for the one after it?